blackbox_repository/utils.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import List, Optional
+import numpy as np
+
+from blackbox_repository.blackbox import Blackbox
+
+
+def metrics_for_configuration(
+        blackbox: Blackbox, config: dict, resource_attr: str,
+        fidelities: Optional[np.array] = None) -> List[dict]:
+    """
+    Returns all results for configuration `config` at fidelities in
+    `fidelities`.
+
+    :param blackbox: Blackbox
+    :param config: Configuration
+    :param resource_attr: Name of resource attribute
+    :param fidelities: Only results at these fidelities are returned.
+        The default is to return results at all fidelity values
+    :return: List of result dicts
+
+    """
+    assert blackbox.fidelity_values is not None, \
+        "Blackbox must come with fidelities"
+    res = []
+    if fidelities is not None:
+        fidelities = set(fidelities)
```

Comment:
This may be slow if you have lots of fidelities (I am running an example with 1K fidelities, you need to do 1K inclusion test in a set of 1K elements). Could you add a comment similar to the one suggested?

```suggestion
        # TODO we test if fidelities belongs to a set for simplicity, in case of performance bottleneck, we could pass
        #  fidelity_min, fidelity_max instead
        fidelities = set(fidelities)
```

Reply:
Uups, you are right, this is in fact pretty bad. I'll change the code.

Reply:
The change you suggest works fine for me.